### ISSUE TITLE
[Bug]: Failed to restore previous session

### ISSUE THREAD

**Issue by peturgq** [All-Hands-AI/OpenHands#2276]
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://opendevin.github.io/OpenDevin/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

Getting error when trying to resume a previous session:
There was a problem when trying to write in your cache folder (/home/enduser/.cache/huggingface/hub). You should set the environment variable TRANSFORMERS_CACHE to a writable directory.

### Current OpenDevin version

```bash
ghcr.io/opendevin/opendevin:0.6
```

### Installation and Configuration

```bash
docker run -it \
    --pull=always \
    -e SANDBOX_USER_ID=$(id -u) \
    -e PERSIST_SANDBOX="true" \
    -e SSH_PASSWORD="xxx" \
    -e WORKSPACE_MOUNT_PATH=$OPENDEVIN_WORKSPACE \
    -v $OPENDEVIN_WORKSPACE:/opt/workspace_base \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -p 3000:3000 \
    --add-host host.docker.internal:host-gateway \
    --name opendevin-app-$(date +%Y%m%d%H%M%S) \
    ghcr.io/opendevin/opendevin:0.6
```

### Model and Agent

- GPT4 Turbo 
- CodeActAgent

### Operating System

Linux

### Reproduction Steps

1. Start opendevin
2. Make it perform some tasks
3. Restart opendevin
4. Error is listed and agent no longer has context of previous chat.

### Logs, Errors, Screenshots, and Additional Context

INFO: browser_env.py:43 - Starting browser env...
14:22:37 - opendevin:ERROR: state.py:64 - Failed to restore state from session: sessions/dbe02db8-bb59-46a5-b6cb-f103d6fc23ca/agent_state.pkl
14:22:37 - opendevin:INFO: agent_controller.py:183 - [Agent Controller dbe02db8-bb59-46a5-b6cb-f103d6fc23ca] Setting agent(CodeActAgent) state from AgentState.LOADING to AgentState.INIT

**Comment by SmartManoj** [comment]
Related issue: https://github.com/lllyasviel/Fooocus/issues/1143

**Comment by enyst** [comment]
@peturgq The default file store is an in-memory file store, which is not persisted if, as you say, you _restarted_ opendevin. I would suggest to try something like this, to save locally. In config.toml:

```
file_store="local"
file_store_path="/absolute/path/to/opendevin/cache/directory"
```
Or whatever path you'd like to keep session files in.

**Comment by SmartManoj** [comment]
> 3\. Restart opendevin

Why are you doing this?

**Comment by assertion** [comment]
@peturgq the reason is, the default value of `jwt_secret` is regenerated after opendevin restarted, in `config.py`, `jwt_secret: str = uuid.uuid4().hex`. So the same token can't generate the same sid after restarted. 

You can set the env parameter JWT_SECRET=A_CONST_VALUT, and sid won't change after restart.

**Comment by tobitege** [comment]
> @peturgq the reason is, the default value of `jwt_secret` is regenerated after opendevin restarted, in `config.py`, `jwt_secret: str = uuid.uuid4().hex`. So the same token can't generate the same sid after restarted.
> 
> You can set the env parameter JWT_SECRET=A_CONST_VALUT, and sid won't change after restart.

Wouldn't a browser cookie be the place to remeber the SID between sessions?

**Comment by assertion** [comment]
> Edit: Could that JWT_SECRET be automatically generated ONCE upon very first build?

The JWT_SECRET is automatically generated right now, but it changed every restart. Not sure whether you mean generate it every time we build the project? And if running the same build, the JWT_SECRET will be the same? @tobitege

**Comment by SmartManoj** [comment]
> > Edit: Could that JWT_SECRET be automatically generated ONCE upon very first build?
> 
> The JWT_SECRET is automatically generated right now, but it changed every restart. Not sure whether you mean generate it every time we build the project? And if running the same build, the JWT_SECRET will be the same? @tobitege

Yes, he meant that only.

**Comment by assertion** [comment]
> generate it every time we build the project? And if running the same build, the JWT_SECRET will be the same

I think this is nice to have feature 😀😀  @SmartManoj @tobitege

**Comment by tobitege** [comment]
Just to clarify: generate only ONCE, on first install. 😀

**Comment by SmartManoj** [comment]
[Don't depend on one secret key](https://supertokens.com/blog/what-is-jwt#:~:text=Dependent%20on%20one%20secret%20key%3A)

**Comment by mamoodi** [comment]
@enyst or @assertion is this still a valid issue?

**Comment by assertion** [comment]
@mamoodi  yes, we may need to modify the generating logic, to the mode @tobitege mentioned above: 

「generate only ONCE, on first install (build)」
